client-default/src/test/scala/com.gu.contentapi.client/GuardianContentClientTest.scala

```diff
@@ -128,4 +129,20 @@ class GuardianContentClientTest extends FlatSpec with Matchers with ScalaFutures
     }
   }
 
+  it should "paginate through all results" in {
+    var pageCount = 0
+    val query = ContentApiClient.search
+      .q("brexit")
+      .fromDate(Instant.parse("2018-05-10T00:00:00.00Z"))
+      .toDate(Instant.parse("2018-05-11T23:59:59.99Z"))
+    // http://content.guardianapis.com/search?q=brexit&from-date=2018-05-10T00:00:00.00Z&to-date=2018-05-11T23:59:59.99Z
+    // has 5 pages of results
+
+    val result = api.paginate(query) { _: SearchResponse => 
+      pageCount += 1 
+      Future.successful(())
+    }
+    
+    result.map { _ => pageCount should be (5) }
```

Comment:
I think that it would be safer to compare the expected total (`SearchResponse#total`) against 1) the number of results returned, and 2) the number of unique ids returned.

Reply:
Since I'm testing pagination and not the underlying query, I think the only thing I'm interested in is that the API correctly goes through all the pages.

Reply:
Sure. But how do we know that it went through _unique_ pages?